db/sqldb/migrations/0003_domain_types.up.sql

```diff
@@ -0,0 +1,242 @@
+    internal_description TEXT NOT NULL,
+    requires_invitation_to_join BOOLEAN NOT NULL,
+    is_accepting_new_members BOOLEAN NOT NULL,
+    is_accepting_new_portfolios BOOLEAN NOT NULL,
+    pacta_version_id TEXT NOT NULL REFERENCES pacta_version (id) ON DELETE RESTRICT,
+    language language NOT NULL,
+    created_at TIMESTAMPTZ NOT NULL
+);
+
+CREATE TABLE initiative_invitation (
+    id TEXT PRIMARY KEY,
+    created_at TIMESTAMPTZ NOT NULL,
+    used_at TIMESTAMPTZ,
+    initiative_id TEXT NOT NULL REFERENCES initiative (id) ON DELETE RESTRICT,
+    used_by_user_id TEXT REFERENCES pacta_user (id) ON DELETE RESTRICT
+);
+
+CREATE TABLE initiative_user_relationship (
+    user_id TEXT NOT NULL,
+    initiative_id TEXT NOT NULL,
+    manager BOOLEAN NOT NULL,
+    member BOOLEAN NOT NULL,
+    updated_at TIMESTAMPTZ NOT NULL,
+    PRIMARY KEY(user_id, initiative_id)
+);
+
+CREATE TYPE file_type AS ENUM (
+    'csv',
+    'yaml',
+    'zip',
+    'html'
+);
+
+CREATE TABLE blob (
+    id TEXT PRIMARY KEY,
+    blob_uri TEXT NOT NULL UNIQUE,
+    file_type file_type NOT NULL,
+    file_name TEXT NOT NULL,
+    created_at TIMESTAMPTZ NOT NULL
+);
+
+-- We use the owner abstraction since there are (at minimum) two types of entities that can own data:
+-- users and initiatives. As we expect that complexity may increase in the future (with organizations)
+-- I think this indirection is worthwhile.
+CREATE TABLE owner (
+    id TEXT PRIMARY KEY,
+    user_id TEXT REFERENCES pacta_user (id) ON DELETE RESTRICT,
+    initiative_id TEXT REFERENCES initiative (id) ON DELETE RESTRICT
+);
+ALTER TABLE owner ADD CONSTRAINT owner_is_always_well_defined CHECK (NUM_NONNULLS(user_id, initiative_id) = 1);
+CREATE INDEX owner_by_user_id ON owner USING btree (user_id);
+CREATE INDEX owner_by_initiative_id ON owner USING btree (initiative_id);
+
+CREATE TYPE failure_code AS ENUM ('UNKNOWN');
+
+CREATE TABLE incomplete_upload (
+    id TEXT PRIMARY KEY,
+    owner_id TEXT NOT NULL REFERENCES owner (id) ON DELETE RESTRICT,
+    admin_debug_enabled BOOLEAN NOT NULL,
+    blob_id TEXT REFERENCES blob (id) ON DELETE RESTRICT, -- Will be deleted when the upload completes (succeeds/fails)
+    name TEXT NOT NULL,
+    description TEXT NOT NULL,    
+    holdings_date TIMESTAMPTZ,
+    created_at TIMESTAMPTZ NOT NULL,
+    ran_at TIMESTAMPTZ,
+    completed_at TIMESTAMPTZ,
+    failure_code failure_code,
+    failure_message TEXT
+);
+
+CREATE TABLE portfolio (
+    id TEXT PRIMARY KEY,
+    owner_id TEXT NOT NULL REFERENCES owner (id) ON DELETE RESTRICT,
+    name TEXT NOT NULL,
+    description TEXT NOT NULL,
+    created_at TIMESTAMPTZ NOT NULL DEFAULT now(),
+    holdings_date TIMESTAMPTZ,
+    blob_id TEXT NOT NULL REFERENCES blob (id) ON DELETE RESTRICT,
+    admin_debug_enabled BOOLEAN NOT NULL,
+    -- These are up for debate, but their basic goal is to help the user identify and differentiate between portfolios.
+    number_of_rows INT8
```

Comment:
Is there a chance that a portfolio contains more than **2 billion** rows? Even at a single byte per line, that's 2 GB of data, which we should be designing especially for if we're expecting that kind of size. If ~2 billion rows max is okay, let's use `INTEGER`, and if not, let's use `BIGINT` (more standard) and chat about requirements.

Reply:
GC fixed.